src/plugins/intel_gpu/src/graph/condition.cpp

```diff
@@ -69,23 +147,67 @@ Condition primitive is resuing memory with the input.
 */
 condition_inst::typed_primitive_inst(network& network, condition_node const& node)
     : parent(network, node),
-      _net_true(network::allocate_network(node.get_program().get_engine(), node.get_branch_true(), true)),
-      _net_false(network::allocate_network(node.get_program().get_engine(), node.get_branch_false(), true)) {
-    auto compare_tensor = node.compare().get_output_layout().get_tensor();
-    auto input_tensor = node.input().get_output_layout().get_tensor();
-    CLDNN_ERROR_TENSOR_SIZES_GREATER_THAN(node.id(),
-                                          "Compare tensor",
-                                          compare_tensor,
-                                          "input tensor",
-                                          input_tensor,
-                                          "Compare primitive is too big.");
-
-    auto compare_with_offster_tensor = compare_tensor + node.offset();
-    CLDNN_ERROR_TENSOR_SIZES_GREATER_THAN(node.id(),
-                                          "Offset with compare tensor",
-                                          compare_with_offster_tensor,
-                                          "input tensor",
-                                          input_tensor,
-                                          "Offset is too big.");
+      _net_true(network::allocate_network(node.get_program().get_engine(), node.get_branch_true().inner_program, true)),
+      _net_false(network::allocate_network(node.get_program().get_engine(), node.get_branch_false().inner_program, true)) {
+    this->set_inner_networks({_net_true, _net_false});
+}
+
+event::ptr condition_inst::execute(const std::vector<event::ptr>& events) {
```

Comment:
Why is it needed? What's wrong with the common primitive_inst::execute() ?

Reply:
I create this custom execute() function to run cldnn::condition without update shape and update impl because the output layout of  condition will be got from inner network's result layout after inner network execution.

At first, I add condition to avoid update shape and impl when current primitive_inst is condition_inst. but the condition code makes complicated code for primitive_inst::execute(). So I created new custom execute() function to run it without update impl/shape.

Reply:
I just don't like that we have a partial duplication of the logic here as the code itself is not that trivial, so ideally we should reuse it as much as possible.
Also, I'm curious if dynamic loop primitive will require similar trick (updating outer primitive shape after completion of body network. If yes, then we definitely need to come up with better solution. 

Overall, I don't fully understand why condition_impl::execute() method can't do update of condition_inst object passed as a parameter. It already updates output_memory, so isn't it just a mater of updating `_impl_params` as well?

Reply:
Okay, I will find better solution for this issue. thanks.

Currently, In condition_impl::execute(), condition_inst update output memory after inner network execution. at latest update, I move the code to update output memory of condition_inst into the condition_impl::execute() because output memory update issue.

Reply:
How about add a pre-stage like "execute_inner_networks()" before update_shape() in common primitive_inst;:execute()?

Reply:
Or I think we can do execute() like 
if (is_dynamic() || !has_inner_networks()) {
    update_shape()
    update_impl()
    update_weight()
} 
...
execute() // update_shape of current inst will be done inside the impl execution of condition & loop.

Reply:
Update codes to remove custom condition_inst::execute(). please review it again. thanks.